pkg/security/security_profile/profile/manager.go

```diff
@@ -511,62 +531,73 @@ func (m *SecurityProfileManager) unlinkProfile(profile *SecurityProfile, workloa
 	seclog.Infof("workload %s (selector: %s) successfully unlinked from profile %s", workload.ID, workload.WorkloadSelector.String(), profile.Metadata.Name)
 }
 
-func (m *SecurityProfileManager) LookupEventOnProfiles(event *model.Event) {
-	evtType := event.GetEventType()
-	if evtType == model.SyscallsEventType || // syscall matching for anomaly detection is already done kernel side
-		evtType == model.FileOpenEventType || evtType == model.BindEventType || // disabled for now
-		evtType == model.ForkEventType || evtType == model.ExitEventType { // no interest in fork/exit events
-		return
-	}
-
-	if event.Error != nil {
-		m.eventFilteringAbsent[evtType].Inc()
+func (m *SecurityProfileManager) LookupEventInProfiles(event *model.Event) {
+	// shortcut for dedicated anomaly detection events
+	if IsAnomalyDetectionEvent(event.GetEventType()) {
+		event.AddToFlags(model.EventFlagsSecurityProfileInProfile)
 		return
 	}
 
-	event.FieldHandlers.ResolveContainerID(event, &event.ContainerContext)
```

Comment:
Unless I'm missing something, I don't think we need it here 🤔 To be safe I added it in the `handleAnomalyDetection` function in case we're looking up a profile by container ID.

Reply:
Yeah, I wasn't sure, I just saw that we had it before and that was removed. It should be fine, I resolve :)